lib/signaling/v2/cancelableroomsignalingpromise.js

```diff
@@ -12,125 +12,140 @@ const {
 
 const { flatMap } = require('../../util');
 
-function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, iceServerSource, encodingParameters, preferredCodecs, options) {
+function createCancelableRoomSignalingPromise(token, wsServer, localParticipant, encodingParameters, preferredCodecs, options) {
   options = Object.assign({
     PeerConnectionManager: DefaultPeerConnectionManager,
     RoomV2: DefaultRoomV2,
     Transport: DefaultTransport
   }, options);
 
-  let transport;
-
-  const PeerConnectionManager = options.PeerConnectionManager;
-  const RoomV2 = options.RoomV2;
-
-  const peerConnectionManager = new PeerConnectionManager(iceServerSource, encodingParameters, preferredCodecs, options);
-
+  const { PeerConnectionManager, RoomV2, Transport, iceServers, log } = options;
+  const peerConnectionManager = new PeerConnectionManager(encodingParameters, preferredCodecs, options);
   const trackSenders = flatMap(localParticipant.tracks, trackV2 => [trackV2.trackTransceiver]);
-
-  peerConnectionManager.setConfiguration(options);
   peerConnectionManager.setTrackSenders(trackSenders);
 
-  const cancelationError = new Error('Canceled');
-
-  return new CancelablePromise(function onCreate(resolve, reject, isCanceled) {
-    peerConnectionManager.createAndOffer().then(function createAndOfferSucceeded() {
-      // NOTE(mmalavalli): PeerConnectionManager#createAndOffer() queues the
-      // initial offer in the event queue for the 'description' event. So,
-      // we are dequeueing to prevent the spurious 'update' message sent by
-      // the client after connecting to a room.
-      peerConnectionManager.dequeue('description');
+  const cancellationError = new Error('Canceled');
 
-      return new Promise((resolve, reject) => {
-        if (isCanceled()) {
-          reject(cancelationError);
-          return;
-        }
-
-        let transportOptions = typeof options.wsServerInsights === 'string'
-          ? { wsServerInsights: options.wsServerInsights }
-          : {};
-
-        if (options.InsightsPublisher) {
-          transportOptions.InsightsPublisher = options.InsightsPublisher;
-        }
+  let transport;
 
-        if (options.NullInsightsPublisher) {
-          transportOptions.NullInsightsPublisher = options.NullInsightsPublisher;
-        }
+  const cancelablePromise = new CancelablePromise((resolve, reject, isCanceled) => {
+    const onIced = iceServers => {
```

Comment:
was moving onIced out of the promise not looking good? I thought it would make this code bit simpler, as onIced was pretty independent (with only complication was its dependency on `cancellationError `)

Reply:
@makarandp0 If I move `onIced` out of the CancelablePromise, then I can't interrupt the events in the callback when the CancelablePromise is cancelled. So, it needs to be defined inside the CancelablePromise.